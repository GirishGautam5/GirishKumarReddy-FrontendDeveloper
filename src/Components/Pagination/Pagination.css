.pagination {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    margin-top: 1rem;
    margin-bottom: 100px;
}

.pagination button {
    width: 40px;
    height: 40px;
    font-family: inherit;
    font-weight: 600;
    font-size: 16px;
    margin: 0 10px;
    border-radius: 6px;
    cursor: pointer;
    transition: all 0.3s ease;
    background: transparent;
    color: #eee;
    border-color: #eee;
    border-width: 3px;
}

.pagination button.active {
    width: 40px;
    height: 40px;
    font-weight: 900;
    border-color: #ffe400;
    background: #ffe400;
    color: #101010;
    border-width: 3px;
}