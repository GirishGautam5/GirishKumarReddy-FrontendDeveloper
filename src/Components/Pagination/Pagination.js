import React from "react";
import "./Pagination.css";

function Pagination(props) {
  const { totalCapsules, capsulesPerPage, currentPage, setCurrentPage } = props;
  let pages = [];

  for (let i = 1; i <= Math.ceil(totalCapsules / capsulesPerPage); i++) {
    pages.push(i);
  }
  return (
    <div className="pagination">
      {pages.map((page, index) => {
        return (
          <button
            key={index}
            onClick={() => setCurrentPage(page)}
            className={page === currentPage ? "active" : ""}
          >
            {page}
          </button>
        );
      })}
    </div>
  );
}

export default Pagination;
