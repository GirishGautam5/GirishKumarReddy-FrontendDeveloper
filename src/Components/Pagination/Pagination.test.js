import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Pagination from "./Pagination";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

describe("Pagination Component test suite", () => {
  it("Pagination Component test case", () => {
    const props = {
      totalCapsules: 25,
      capsulesPerPage: 9,
      currentPage: 1,
      setCurrentPage: jest.fn(),
    };
    const paginationComponent = shallow(<Pagination {...props} />);
    expect(toJson(paginationComponent)).toMatchSnapshot();
  });
});
