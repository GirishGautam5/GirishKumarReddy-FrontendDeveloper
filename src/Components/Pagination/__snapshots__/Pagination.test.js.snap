// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component test suite Pagination Component test case 1`] = `
<div
  className="pagination"
>
  <button
    className="active"
    key="0"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className=""
    key="1"
    onClick={[Function]}
  >
    2
  </button>
  <button
    className=""
    key="2"
    onClick={[Function]}
  >
    3
  </button>
</div>
`;
